### PR TITLE
Feature/plantation

### LINKED ISSUE #4864: Healer requests things he already has
### Minecolonies version
Version: 0.11.917 Alpha

### Expected behavior

Healer does request items he is lacking and uses what he has

### Actual behaviour

Healer is constantly requesting items that he has plenty of and does nothing with. He also does not heal people when he does that, instead waits for you to bring him more. Moving stacks around in his inventory sometimes helps.

### Steps to reproduce the problem

1. Take a larger colony
2. Build a hospital
3. Wait for several people to get sick and be annoyed by the healer requesting stuff he has stacks of all the time

### Logs: Latest.log and crashlog at https://gist.github.com/
No logs

### LINKED ISSUE #4946: Farmer asking for hoe for ages and stucked
### Minecolonies version

Minecolonies version 0.11.933-ALPHA
Minecraft version 1.15.2
Forge version 3.11.77

### Expected behavior

- Farmer go to work with its hoe
- No more asking for hoe

### Actual behaviour

- Farmer is asking for Hoe. We can fullfill request but still show the request
- Request is not cancellable
- The crafter request is not cancellable either
- See screenshots

### Steps to reproduce the problem

1. Have a farmer and a crafter in a colony
2. Let the farmer do its job and wait until it asks for a hoe
3. Wait the the crafter does the craft (it has sticks and stone in his inventory, same in hut, and there is the materials in warehouse also)
4. Nothing happens. So fullfill the request. The hoe is in the farmer inventory… However, still asking for a hoe and does not do anything
5. Try to cancel requests, none of the 2 requests in screenshot are cancellable

6. I tried to fire the farmer and free its inventory, assign another worker, and same behavior

### Logs: Latest.log and crashlog at https://gist.github.com/

- latest.log:
- crashlog:

### LINKED ISSUE #4955: Fisherman is broken
minecolonies-0.11.937-ALPHA-universal.jar

Building is Nordic lvl1. Fisherman is trying to fish from inside of the building, waisting the fishing rod. Sometimes, he goes outside in the water. He rarely succeeds to catch fish.

### LINKED ISSUE #4956: Crash on death of deliveryman
### Minecolonies version
Version: 0.11.937-ALPHA

### Expected behavior
No crash

### Actual behaviour
My world is in a state of defending a barbarian invation, and the game crashes every time I load that save.
Reading the crash log it seems to happen becouse a delivery man get killed by a barbarian and the request system tryes to reasign active requests or something like that.

### Steps to reproduce the problem
Hard to say for sure. Kill a delivery person with active requests?

### Logs: Latest.log and crashlog at https://gist.github.com/
https://gist.github.com/SirSmuggler/4d051a012495f23e0151d8eead6c56f0

### LINKED ISSUE #4958: Builder crash
# Minecolonies version

Version: 0.11.932-ALPHA

### Expected behavior
No crash
+
second problem : my 2 builders have bot issues with the request system, and with their own inventory/hut inventory (when dropping an item in it it's not recognized)
### Actual behaviour
crash
+
builder never being delivered by my deliveryman

### Steps to reproduce the problem

wait for a building to be constructed, it doesn't happend everytime, tho the 2nd problem (not the crash) happend a lot since 4 days (I try to keep up with each important alpha - useful bugfixe -, might be related to a change in one of them, didn't had this behaviour 4 days ago)

### Logs: Latest.log and crashlog at https://gist.github.com/

- latest.log: https://gist.github.com/Lyacs/626b50f151b56116fd98d651d7e3b98f
- crashlog: https://gist.github.com/Lyacs/b852f4c43179720931e1c42075dbf519

Thank you very much for all the support you provide with the mod, keep up like that !

### LINKED ISSUE #4966: Farmer won't plant pumpkin and melon
### Minecolonies version

Version:  minecolonies-0.11.939-ALPHA-universal

### Expected behavior

- Farmer should plant pumpkin and melon seeds, but won't.

### Actual behaviour

- Farmer hoes the field and then runs around like he would plant the seeds, but does not. I tried restarting the worker and rehiring him, behaviour is still the same. When i plant the seeds myself he will harvest both.

### Steps to reproduce the problem

1. Build field
2. Set melon or pumkin
3. Provide seeds
4. Field gets hoed
5. Farmer just stands around not planting seeds

### Logs: Latest.log and crashlog at https://gist.github.com/

- logs: no related problems or excetions shown

### LINKED ISSUE #4974: Current Activity: NEEDS_ITEM
minecolonies-0.11.940-ALPHA-universal

I have two workers, miner and farmer stuck like this for many ingame days and i have no ide a why. The yare just standing next to their building controller thingy.

They do not ask for any item, but the "/mc citizen info" says they need something, but does'nt say what.

Can you make it so it will just say what item they need in the command output ?

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #4946
Closes #4956
Closes #4955
Closes #4958
Closes #4864 
Closes #4974
Closes #4966

# Changes proposed in this pull request:
- Add Plantation worker (Sugarcane, cactus, bamboo) Crafts paper + books
- Fix tool request issues
- Fix fulfillment issues if request is in retrying

Review please

### CODE REVIEW

**Comment by LDTTeam-Buildsystem** [comment]
This PR has been automatically built and hosted by our build system, during this process there was a crash, error, or otherwise, please check on this at: https://teamcity.minecolonies.com/viewLog.html?buildId=39708&buildTypeId=LetSDevTogether_Minecolonies_PullRequests_2_DeployTestServer

File: src/api/java/com/minecolonies/api/colony/requestsystem/requestable/Tool.java [hunk 1]
```diff
@@ -142,7 +142,7 @@ public boolean matches(@NotNull final ItemStack stack)
                 && getToolClasses(stack).stream()
                 .filter(s -> getToolClass().getName().equalsIgnoreCase(s))
                 .map(ToolType::getToolType)
-                .anyMatch(t -> t != ToolType.NONE && !(stack.isDamaged() || stack.getDamage() > 0) && ItemStackUtils.hasToolLevel(stack, t, getMinLevel(), getMaxLevel()));
+                .anyMatch(t -> t != ToolType.NONE && (stack.getDamage() > 0 || !stack.isDamaged()) && ItemStackUtils.hasToolLevel(stack, t, getMinLevel(), getMaxLevel()));
```
Comment (someaddons) [inline src/api/java/com/minecolonies/api/colony/requestsystem/requestable/Tool.java:145]: nice find :)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowHutPlantation.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import com.minecolonies.coremod.colony.buildings.workerbuildings.BuildingMiner;
+import com.minecolonies.coremod.colony.buildings.workerbuildings.BuildingPlantation;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * Window for the crusher hut.
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowHutPlantation.java]: old javadoc

File: src/main/java/com/minecolonies/coremod/client/gui/WindowHutPlantation.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+{
+    /**
+     * The mode button id.
+     */
+    private static final String BLOCK_BUTTON = "block";
+
+    /**
+     * The id of the gui.
+     */
+    private static final String CRUSHER_RESOURCE_SUFFIX = ":gui/windowhutplantation.xml";
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowHutPlantation.java]: still not the crusher :)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowHutPlantation.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    /**
+     * Constructor for the window of the miner hut.
+     *
+     * @param building {@link BuildingMiner.View}.
+     */
+    public WindowHutPlantation(final BuildingPlantation.View building)
+    {
+        super(building, Constants.MOD_ID + CRUSHER_RESOURCE_SUFFIX);
+        final Button crushingSettingsButton = findPaneOfTypeByID(BLOCK_BUTTON, Button.class);
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowHutPlantation.java]: button isnt a crusher either

File: src/main/java/com/minecolonies/coremod/client/gui/WindowHutPlantation.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        registerButton(BLOCK_BUTTON, this::switchCrushingMode);
+        setupSettings(crushingSettingsButton);
+    }
+
+    /**
+     * Switch the mode after clicking the button.
+     *
+     * @param crushingSettingsButton the clicked button.
+     */
+    private void switchCrushingMode(final Button crushingSettingsButton)
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowHutPlantation.java]: plantingmodes?^^

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingPlantation.java [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.minecolonies.api.util.constant.BuildingConstants.CONST_DEFAULT_MAX_BUILDING_LEVEL;
+import static com.minecolonies.api.util.constant.NbtTagConstants.*;
+import static com.minecolonies.api.util.constant.ToolLevelConstants.TOOL_LEVEL_WOOD_OR_GOLD;
+
+/**
+ * Class of the plantation building.
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingPlantation.java:47]: little description of what he's doing here would be nice

**Comment by LDTTeam-Buildsystem** [comment]
This PR has been automatically built and hosted by our build system, during this process there was a crash, error, or otherwise, please check on this at: https://teamcity.minecolonies.com/viewLog.html?buildId=39784&buildTypeId=LetSDevTogether_Minecolonies_PullRequests_2_DeployTestServer

**Review by marchermans — DISMISSED** [review]

**Comment by Perlo0ung** [comment]
@Raycoms : small question, which of the workers does make sugar out of sugarcane, shouldn't that also be a thing of the plantation worker?

**Comment by Raycoms** [comment]
ye it should

**Comment by Perlo0ung** [comment]
want me to create an issue for that?

**Comment by Raycoms** [comment]
Sure

**Pull request merged**